invenio_rdm_records/services/generators.py

```diff
@@ -117,6 +117,21 @@ def query_filter(self, **kwargs):
             return q_public
 
 
+class IfRestrictedForCommunity(IfRestricted):
```

Comment:
minor: the name is a bit ambiguous, it is not clear what is restricted for the community.

```suggestion
class IfFilesRestrictedForCommunity(IfRestricted):
```

However I also understand that we're using like `IfFilesRestrictedForCommunity("files")` and "files" becomes redundant.